### DISCUSSION TITLE
Question on ICC calculation Bayesian models

### DISCUSSION THREAD

**Discussion by moosterwegel** [easystats/performance#438] [Q&A] [answered]
First of all, thanks for the great package and documentation. 

The `variance_decomposition` function uses the posterior predictive distribution to calculate the variance components for brms models to obtain an ICC like measure. However, if I look into the ICC literature I don't see the posterior predictive distribution mentioned. Nakagawa (2010)<sup>1</sup> for example only seems to mention the posterior distribution for Bayesian models and in this [blogpost](https://rpsychologist.com/GLMM-part1-lognormal)<sup>2</sup> from Kristoffer Magnusson they also (simply) use draws from the posterior distribution for the calculation of the variance components. So I was wondering what the rational for the posterior predictive distribution is? 

Could very well be that I'm missing something obvious. Any pointers are appreciated. Thanks.

1. Nakagawa S, Schielzeth H. Repeatability for Gaussian and non-Gaussian data: a practical guide for biologists. Biological Reviews. 2010;85(4):935–56.  [doi.org/10.1111/j.1469-185X.2010.00141.x ](https://doi.org/10.1111/j.1469-185X.2010.00141.x)
2. https://rpsychologist.com/GLMM-part1-lognormal

**Comment by strengejacke** [comment, marked as answer]
See https://discourse.mc-stan.org/t/rstanarm-extracting-variance-components/4409/4

This was a proposal by one of the Stan developers.

**Reply by bwiernik** [reply]
As a general principle, with an MCMC or bootstrapped model, any derived quantities are best estimated by pushing through data to construct a predictive distraction and computing from there. Other methods for estimating derived quantities and their sampling errors are only needed when there isn't inherently a set of predictions with the correlation structure of the model parameters baked in
